File: lua/p7.lua
function isPrime(n)--will check if n is prime
    lim = math.sqrt( n ) --sqrt of n is the largest possible factor pair of n
    for i = 2, lim - 1 do 
        if(n % i == 0) then
            return false
        end
    end
    return true --no factors found, is prime
end

counter = 1--start at one so currentN can start at 3
--this allows me to increment my check by 2 as 2 is the only prime
--this halfs search time
currentN = 3
while(counter < 10000) do
    if(isPrime(currentN)) then
        counter = counter + 1
    end

    currentN = currentN + 2
end
print(currentN - 3)

File: python/p7.py
# -*- coding: utf-8 -*-
"""
Created on Tue May 15 11:23:48 2018

@author: afielding
"""
import math
def isPrime(n):
    s = int(math.sqrt(n))
    for x in range(2, s + 1):
        if(n % x == 0):
            return False
    return True

x = 0
y = 1
while (x <= 10000):
    y += 1
    if(isPrime(y)):
        x += 1
print(y)